Promote feature gate EndpointSlice to GA (#5393)
antrea-io/antrea: docs/feature-gates.md

Reviewer: Please update the `EndpointSlice` and `TopologyAwareHints` section for the change.
I suppose `- EndpointSlice is enabled` can be removed for `TopologyAwareHints`?

Reviewer: Not addressed? I think we can update the line `EndpointSlice is enabled.` as below?
```
EndpointSlice API version v1 is supported and set as enabled in Kubernetes
```

#### Requirements for this Feature

- `AntreaProxy` is enabled.

> - EndpointSlice API version v1 is supported and set as enabled in Kubernetes.

Reviewer 2: ```suggestion
- EndpointSlice API version v1 is available in Kubernetes.
```
